### ISSUE TITLE
Report conflicting explicit bindings

### ISSUE THREAD

**Issue by nbenty** [shader-slang/slang#93]
Not Slang's fault, but it would be great if it will report conflicts in external bind locations.
For example:
```C++
layout(set = 0, binding = 0) uniform texture2D gTexture;
layout(set = 0, binding = 0) uniform sampler gSampler;
```

The shader compiles successfully. I don't know if it's a valid GLSL shader or not, perhaps the correct place for this issue is glslang.

**Comment by tangent-vector** [comment]
This has some overlap with #38.

I'd like to make this a warning (two parameters with different names and overlapping bindings), with an option to make it trigger an error. This hasn't bubbled up to the top of the priority list yet.

As far as I understand, this is valid GLSL. It is okay to define overlapping bindings, so long as a given compiler shader ("version") only uses one or the other (I don't know if that is a static constraint or a dynamic one).

**Comment by nbenty** [comment]
Is it valid even if the declarations are in the same shader?
Issue #96 both the constant-buffer and sampler are bound to the same location and are being used in the shader, I don't see why this is valid.

I'm not suggesting Slang is doing anything wrong. Either glslang has a bug or GLSL has a bug.

**Comment by tangent-vector** [comment]
Do we get errors reported from the driver?

I would need to comb the spec carefully to know what is or isn't valid.

**Comment by tfoleyNV** [comment]
I implemented a basic version of this in #118. I'll close this if/when I get confirmation that it seems to meet requirements (and doesn't give false positives).

**Comment by tfoleyNV** [comment]
I've seen this trigger now in cases where it should, and nobody has reported false positives (yet), so I'm going to consider this good.
